lib/Basics/GlobalSerialization.h

```diff
@@ -0,0 +1,82 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2023-2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Max Neunhoeffer
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
```

Comment:
```suggestion
#pragma once

#if ARANGODB_ENABLE_FAILURE_TESTS
```

Reply:
Good suggestion, this will enforce that such global sync points are only in failure tests! Will chang.

Reply:
It turns out that this is not possible. The thing is that `TRI_IF_FAILURE` is defined to be
```
#define TRI_IF_FAILURE(what) if constexpr (false)
```
if failure tests are not compiled in. That means that the code in this block must still compile, even if failure tests are not compiled in. Therefore we cannot hide the headers in this case. Reverting change.